daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -31,7 +31,7 @@ message ContractInstance {
    string package_name = 4; // required *since dev*
     com.daml.lf.value.Identifier template_id = 1;
     com.daml.lf.value.VersionedValue arg_versioned = 2;
-    string agreement = 3;
+    reserved 3; // was agreement
```

Comment:
If you touch the proto, you will have to modify the doc as well. `/daml-lf/spec/transaction.rst`

Reply:
Better not touch the proto and let this change in the scope of https://github.com/digital-asset/daml/issues/18240 (with a todo)